src/interpreter.c

```diff
@@ -244,7 +259,19 @@ static val_t i_eval_call(struct v7 *v7, struct ast *a, ast_off_t *pos, val_t sco
   end = ast_get_skip(a, *pos, AST_END_SKIP);
   ast_move_to_children(a, pos);
   v1 = i_eval_expr(v7, a, pos, scope);
-  if (!v7_is_function(v1)) {
+
+  if (v7_is_cfunction(v1)) {
+    char buf[20];
+    int n, i;
+    val_t args = v7_create_value(v7, V7_TYPE_ARRAY_OBJECT);
+    for (i = 0; *pos < end; i++) {
+      res = i_eval_expr(v7, a, pos, scope);
+      v7_array_append(v7, args, res);
+      n = snprintf(buf, sizeof(buf), "%d", i);
+      v7_set_property_value(v7, args, buf, n, 0, res);
```

Comment:
oops :-)
overlook. removing v7_array_append() call from here, cause it is slower, and we can have C function calls done in loops, so we'd rather be fast.